lib/Common/Memory/Recycler.h

```diff
@@ -152,25 +150,6 @@ struct InfoBitsWrapper{};
 #define RecyclerNewWithBarrierFinalizedClientTracked(recycler,T,...) static_cast<T *>(static_cast<FinalizableObject *>(AllocatorNewBase(Recycler, recycler, AllocFinalizedClientTrackedWithBarrierInlined, T, __VA_ARGS__)))
 #endif
 
-
-#if defined(RECYCLER_WRITE_BARRIER_ALLOC) && GLOBAL_ENABLE_WRITE_BARRIER
-#define RecyclerNew                         RecyclerNewWithBarrier
-#define RecyclerNewPlus                     RecyclerNewWithBarrierPlus
-#define RecyclerNewPlusZ                    RecyclerNewWithBarrierPlusZ
-#define RecyclerNewZ                        RecyclerNewWithBarrierZ
-#define RecyclerNewStruct                   RecyclerNewWithBarrierStruct
-#define RecyclerNewStructZ                  RecyclerNewWithBarrierStructZ
-#define RecyclerNewStructPlus               RecyclerNewWithBarrierStructPlus
-#define RecyclerNewArray                    RecyclerNewWithBarrierArray
-#define RecyclerNewArrayZ                   RecyclerNewWithBarrierArrayZ
-#define RecyclerNewFinalized                RecyclerNewWithBarrierFinalized
-#define RecyclerNewFinalizedPlus            RecyclerNewWithBarrierFinalizedPlus
-#define RecyclerNewTracked                  RecyclerNewWithBarrierTracked
-#define RecyclerNewEnumClass                RecyclerNewWithBarrierEnumClass
-#define RecyclerNewWithInfoBits             RecyclerNewWithBarrierWithInfoBits
-#define RecyclerNewFinalizedClientTracked   RecyclerNewWithBarrierFinalizedClientTracked
-#endif
```

Comment:
This breaks the plugin? The plugin relies on allocFunc name to tell if a type is allocated in writebarrier memory (allocFunc name contains "WithBarrier", or "InfoBitsWrapper" contains "WithBarrierBit").

If you really want this change, maybe we can change the plugin: any "Recycler" new without "Leaf" in allocFunc name, or any "InvoBitsWrapper" variant recycler new, the type must be annotated. Is this precise?